### PR TITLE
Remove bash dep from init scripts

### PR DESCRIPTION
**Pull request by am11**
upstream changes https://github.com/dotnet/arcade/pull/14500

### CODE REVIEW

**Review by ViktorHofer — APPROVED** [review]

File: eng/common/native/init-distro-rid.sh [hunk 1]
```diff
@@ -11,67 +11,66 @@
...
+                nonPortableRid="omnios.$__omnios_major_version-${targetArch}"
+                ;;
             joyent_*)
-                __smartos_major_version=$(echo "${__uname_version:7:4}")
-                nonPortableRid=smartos."$__smartos_major_version"-"$targetArch"
-            ;;
-            illumos_*)
-                nonPortableRid=openindiana-"$targetArch"
-            ;;
+                __smartos_major_version=$(echo "$__uname_version" | cut -c9-10)
```
Comment (hoyosjs) [inline eng/common/native/init-distro-rid.sh:57]: Is this a typo? Or is the range also 9-10?

File: eng/common/native/init-distro-rid.sh [hunk 2]
```diff
@@ -11,67 +11,66 @@
...
+        # shellcheck disable=SC1091
         if [ -e "${rootfsDir}/etc/os-release" ]; then
-            source "${rootfsDir}/etc/os-release"
-
-            if [[ "${ID}" == "rhel" || "${ID}" == "rocky" || "${ID}" == "alpine" ]]; then
-                # remove the last version digit
-                VERSION_ID="${VERSION_ID%.*}"
+            . "${rootfsDir}/etc/os-release"
+            if [ "${ID}" = "rhel" ] || [ "${ID}" = "rocky" ] || [ "${ID}" = "alpine" ]; then
+                VERSION_ID="${VERSION_ID%.*}" # Remove the last version digit for these distros
```
Comment (hoyosjs) [inline eng/common/native/init-distro-rid.sh:24]: We stopped using it this trimming type of expansion for __solaris_major_version but not here. Is there difference in shell supports?

File: eng/common/native/init-distro-rid.sh (hunk 1 above)
Comment (am11) [reply eng/common/native/init-distro-rid.sh:57]: Bash syntax was 'skip 8 chars and select next two', new syntax is 'cut/select characters 9 and 10'.

File: eng/common/native/init-distro-rid.sh (hunk 2 above)
Comment (am11) [reply eng/common/native/init-distro-rid.sh:24]: Ah, ChatGPT thinks this is the way to go, and shellcheck doesn't complain.
Bourne Shell:
image


But good eye! I'll make it consistent later today.

Comment (am11) [reply eng/common/native/init-distro-rid.sh:24]: ChatGPT's reply:
> If your target environments are systems that are 15 years old or newer, you are generally in a safe zone regarding the support for ${variable%pattern} parameter expansion. This feature is part of the POSIX standard, and most Unix-like systems from the last 15 years would have /bin/sh linked to a shell that supports POSIX parameter expansion

so advanced ops like substring extraction `${string:position:length}` are bash-specific. So I think we can safely use that parameter expansion syntax for now.

**Pull request merged**